File: problems/problem 1/test_cases.java
public class test_cases {
    private static int failures = 0;

    private static void check(String input, String expected) {
        String output = solution.firstNonRepeating(input);
        if (output.equals(expected)) {
            System.out.println("PASS: " + input);
        } else {
            System.out.println("FAIL: " + input + " Expected=" + expected + " Got=" + output);
            failures++;
        }
    }

    public static void main(String[] args) {
        check("leetcode", "l");
        check("aabb", "None");
        check("abcabcde", "d");
        check("aabbccddeeff", "None");
        check("swiss", "w");

        if (failures > 0) System.exit(1);
        else System.exit(0);
    }
}

File: problems/problem 1/test_cases.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>


char* firstNonRepeating(char *s);

int failures = 0;

void check(char *input, char *expected) {
    char *output = firstNonRepeating(input);
    if (strcmp(output, expected) == 0) {
        printf("PASS: %s\n", input);
    } else {
        printf("FAIL: %s Expected=%s Got=%s\n", input, expected, output);
        failures++;
    }
}

int main() {
    check("leetcode", "l");
    check("aabb", "None");
    check("abcabcde", "d");
    check("aabbccddeeff", "None");
    check("swiss", "w");

    if (failures > 0) return 1;
    return 0;
}
